contrib/wheels/build_wheel.py

```diff
@@ -0,0 +1,118 @@
+import os
+import re
+import sys
+import platform
+import subprocess
+import multiprocessing
+import shutil
+import glob
+
+from setuptools import setup, Extension
+from setuptools.command.build_ext import build_ext
+from distutils.version import LooseVersion
+
+
+class CMakeExtension(Extension):
+    def __init__(self, name, sourcedir=''):
+        Extension.__init__(self, name, sources=[])
+        self.sourcedir = os.path.abspath(sourcedir)
+
+
+class CMakeBuild(build_ext):
+    def run(self):
+        try:
+            out = subprocess.check_output(['cmake', '--version'])
+        except OSError:
+            raise RuntimeError(
+                "CMake must be installed to build the following extensions: " +
+                ", ".join(e.name for e in self.extensions))
+
+        if self.is_windows():
+            cmake_version = LooseVersion(
+                re.search(r'version\s*([\d.]+)', out.decode()).group(1))
+            if cmake_version < '3.1.0':
+                raise RuntimeError("CMake >= 3.1.0 is required on Windows")
+
+        for ext in self.extensions:
+            self.build_extension(ext)
+
+    @staticmethod
+    def is_windows():
+        tag = platform.system().lower()
+        return tag == "windows"
+
+    @staticmethod
+    def is_linux():
+        tag = platform.system().lower()
+        return tag == "linux"
+
+    def build_extension(self, ext):
+        extdir = os.path.abspath(
+            os.path.dirname(self.get_ext_fullpath(ext.name)))
+        if not os.path.isdir(extdir):
+            os.mkdir(extdir)
+
+        # test env variable to determine whether to build in debug
+        if os.environ.get("DEBUG") is not None:
+            cfg = 'Debug'
+        else:
+            cfg = 'Release'
+        build_args = ['--config', cfg]
+
+        cpu_count = max(2, multiprocessing.cpu_count() // 2)
+        build_args += ['--', '-j{0}'.format(cpu_count)]
+
+        # call configure
+        # default install everything?
+        solvers = ["btor"] #, "cvc4", "msat"]
+        solver_path = {"btor": "boolector", "cvc4": "CVC4", "msat": "mathsat"}
+        root_path = os.path.dirname(os.path.dirname(os.path.abspath(os.path.dirname(__file__))))
+        build_dir = os.path.join(root_path, "build")
+
+        for solver in solvers:
+            solver_dir = os.path.join(root_path, "deps", solver_path[solver])
```

Comment:
The configure script allows specifying different solver locations. I don't think that will work for this script but that's probably okay. This only needs to be run occasionally by a developer, right?

Reply:
Correct. This script is not intended to be run regularly during development process. It is written for CI to build wheels.